src/platforms/linux/daemon/dbusservice.cpp

```diff
@@ -40,6 +40,14 @@ DBusService::DBusService(QObject* parent) : Daemon(parent) {
 
 DBusService::~DBusService() { MVPN_COUNT_DTOR(DBusService); }
 
+WireguardUtilsLinux* DBusService::wgutils() {
+  if (!m_wgutils) {
+    m_wgutils = new WireguardUtilsLinux();
+    m_wgutils->setParent(this);
```

Comment:
This works too, but why not passing the argument in the CTOR?

Reply:
The issue with this approach is that we can forget to call setParent().

Reply:
I agree can forget to call setParent. In the other case couldn't I forget to call delete? Just parenting to QObject (via WireguardUtils) doesn't ensure object is destroyed (or maybe I got something wrong when I tried this before). Or are you thinking I make the parent Daemon in WireguardUtils.

Reply:
Or is the idea that when I call `new WireguardUtilsLinux`, I call it with `new WireguardUtilsLinux(this)` where this is dbusservice and that way I'm forced to call use a `this` so I won't forget.